File: SEM 4/ADS Lab/CLASS 1 09-03-2022/factorial.cpp

#include <iostream>

using namespace std;

int fact(int x)
{
    if(x<=1) return 1;
    else return x*fact(x-1);
}

int main()
{
    int n;
    cout<<"Enter the number = ";
    cin >>n;
    
    cout<<fact(n);

    return 0;
}

File: SEM 5/WEB TECH LAB (using java)/SESSION 2 24-08-2022/Factorial.java

import java.util.*;

public class Factorial {
    int fact(int n)
    {
        if(n<2) return 1;
        else return n*fact(n-1);
    }
    
    public static void main(String[] args)
    {
        Scanner inp=new Scanner(System.in);
        System.out.print("Enter the number - ");
        int num=inp.nextInt();
        
        Factorial obj=new Factorial();
        System.out.println("Factorial = "+obj.fact(num));
    }
}
